Makefile

```diff
@@ -33,22 +42,27 @@ test:  ## Run all tests
 test-unit:  ## Run unit tests
 	go test ./...
 
-# Run Cypress locally against dockerised dev server
-test-e2e-dev:
+test-e2e-dev: e2e-dependencies  ## Run Cypress locally against dockerised dev server
 	if ! docker compose -p $(DEV_PROJECT_NAME) ps --services --filter "status=running" | grep nginx; then \
 		echo "dev environment not running"; exit 1; \
 	fi
 	cd web && CYPRESS_baseUrl=http://localhost:8000 npx cypress run --headless --browser chrome
 
-## Run cypress against the release build
-test-e2e-release:
+test-e2e-release: e2e-dependencies ## Run cypress against the release build
 	cd e2e && \
```

Comment:
Local test failures resulted in the test container remaining for me. So perhaps something like this to delete the containers even on test failures:

```
test-e2e-release: e2e-dependencies ## Run cypress against the release build
	cd e2e && \
	docker compose -p $(E2E_PROJECT_NAME) build && \
	( docker compose -p $(E2E_PROJECT_NAME) run --rm cypress; EXIT_CODE=$$?; docker compose -p $(E2E_PROJECT_NAME) down --remove-orphans; exit $$EXIT_CODE )
```

Reply:
for me this is the desired behaviour – if something breaks I want to look at the logs